File: src/main/java/Characters/Direction.java
package Characters;

public enum Direction {
	// JavaFXの画面座標は「上がマイナスの値、下がプラスの値」になっている
	
	// 上下左右に1ピクセル移動
	UP(0,-1),
	DOWN(0,1),
	LEFT(-1,0),
	RIGHT(1,0),
	// 押されていない停止状態。始まるまで仙石さんを停止させるため
	NONE(0,0);
	
	private final double dx;
	private final double dy;
	
	// コンストラクタ
	Direction(double dx, double dy){
		this.dx = dx;
		this.dy = dy;
	}
	
	// 移動した量を取得するゲッター
	public double getDX() {return dx;}
	public double getDY() {return dy;}
	
	
	// JavaFXのキー入力(KeyCode)から対応するDirection(動き)を返すメソッド
	public static Direction fromKeyCode(javafx.scene.input.KeyCode code) {
		return switch(code) {
		// 矢印上
		case UP -> UP;		
		// 矢印下
		case DOWN -> DOWN;
		// 矢印左
		case LEFT -> LEFT;
		// 矢印右
		case RIGHT -> RIGHT;
		// それ以外のキー
		default -> NONE;
		
		};
	}
}

File: src/main/java/Characters/Direction.js

export const Direction = Object.freeze({
    UP:    { name: "UP",    dx:  0, dy: -1 },
    DOWN:  { name: "DOWN",  dx:  0, dy:  1 },
    LEFT:  { name: "LEFT",  dx: -1, dy:  0 },
    RIGHT: { name: "RIGHT", dx:  1, dy:  0 },
    NONE:  { name: "NONE",  dx:  0, dy:  0 }
});

// JavaFXの fromKeyCode の代わり（ブラウザのKeyboardEvent.keyを受け取る）
export function directionFromKey(key) {
    switch (key) {
        case "ArrowUp":    case "w": case "W": return Direction.UP;
        case "ArrowDown":  case "s": case "S": return Direction.DOWN;
        case "ArrowLeft":  case "a": case "A": return Direction.LEFT;
        case "ArrowRight": case "d": case "D": return Direction.RIGHT;
        default: return Direction.NONE;
    }
}